tools/generate-deps-configmaps/generate.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See README.md for usage information
+
+import os
+import sys
+import yaml
+import argparse
+from string import Template
+
+
+# Get a list of templates in the directory. We remove the suffix here as the name of the
+# template is also used in the name of the ConfigMap.
+def get_templates(path):
+    EXTENSION = ".template"
+    templates = []
+    for _, _, files in os.walk(path):
+        for file in files:
+            if file.endswith(extension):
+                templates.append(file.replace(extension, ""))
```

Comment:
sorry, the previous suggestion was a partial which breaks things if accepted as it was. that's on me
```suggestion
            if file.endswith(EXTENSION):
                templates.append(file.replace(EXTENSION, ""))
```